app/controllers/wbs_controller.rb

```diff
@@ -13,10 +14,20 @@ def index
         render :layout => !request.xhr?
       }
       format.api  {
-        @issues = @project.issues.visible
-                    .where
-                      .not(tracker_id: RedmineWbs.excluded_tracker_ids.reject { |c| c.empty? })
-                    .order("#{Issue.table_name}.root_id ASC, #{Issue.table_name}.lft ASC")
+        query = "
+          SELECT I.*
+            FROM issues I
+           WHERE I.`project_id` = ?
+             AND I.`root_id` NOT IN (
+              SELECT I.`id`
+                FROM `issues` I
+                JOIN `trackers` T ON t.`id` = I.`tracker_id`
+               WHERE I.`root_id` = I.`id` AND I.`tracker_id` IN (?) OR T.`fields_bits` & 64
+            )
+          ORDER BY I.`root_id` ASC, I.`lft` ASC
+        "
```

Comment:
Two things:
1. We lost the `.visible` on the query
2. The current query product a full scan on the issue table. We should optimize it with a query like this one:
```
SELECT I.*
FROM issues I
WHERE I.`root_id` IN (
    SELECT I.`id`
    FROM `issues` I
    WHERE I.`root_id` = I.`id` 
        AND I.`project_id` = ? 
        AND I.`tracker_id` NOT IN (SELECT T.`id` FROM `trackers` T WHERE T.`id` IN (?) OR T.`fields_bits` & 64)
)
ORDER BY I.`root_id` ASC, I.`lft` ASC
```

3. Could you replace the `64` with a variable that explain what it is.